cmd/bbolt/surgery_commands.go

```diff
@@ -241,3 +243,69 @@ at dstPageId in DST.
 The original database is left untouched.
 `, "\n")
 }
+
+// clearPageCommand represents the "surgery clear-page" command execution.
+type clearPageCommand struct {
+	*surgeryCommand
+}
+
+// newClearPageCommand returns a clearPageCommand.
+func newClearPageCommand(m *surgeryCommand) *clearPageCommand {
+	c := &clearPageCommand{}
+	c.surgeryCommand = m
+	return c
+}
+
+// Run executes the command.
+func (cmd *clearPageCommand) Run(args ...string) error {
+	// Parse flags.
+	fs := flag.NewFlagSet("", flag.ContinueOnError)
+	help := fs.Bool("h", false, "")
+	if err := fs.Parse(args); err != nil {
+		return err
+	} else if *help {
+		fmt.Fprintln(cmd.Stderr, cmd.Usage())
+		return ErrUsage
+	}
+
+	if err := cmd.parsePathsAndCopyFile(fs); err != nil {
+		return fmt.Errorf("clearPageCommand failed to parse paths and copy file: %w", err)
+	}
+
+	// Read page id.
+	pageId, err := strconv.ParseUint(fs.Arg(2), 10, 64)
+	if err != nil {
+		return err
+	}
+
+	// Read the page
+	p, buf, err := guts_cli.ReadPage(cmd.dstPath, pageId)
+	if err != nil {
+		return fmt.Errorf("ReadPage failed: %w", err)
+	}
+
+	pageInfo := p.String()
+
+	// Update and rewrite the page
+	p.SetCount(0)
+	p.SetOverflow(0)
+	if err := guts_cli.WritePage(cmd.dstPath, buf); err != nil {
+		fmt.Errorf("WritePage failed: %w", err)
+	}
```

Comment:
Why not make it make it another command in `surgeon` ? 

I imagine it exactly as a place to land this types of 'operations' and CLI should be free of deep business logic assumptions.

Reply:
Do you mean command `surgery write-page`?

Reply:
I think that we should put the logic within this file: 

[internal/surgeon/surgeon.go](https://github.com/etcd-io/bbolt/pull/389/files#diff-e534fe0bacede64743f58cb16c053ab3183e64ea54afb20166373575c27f7c5c)

So have there a method: 

```
func ClearPage(path string, pgid pgId) error {...}
```

and call it from the CLI rather than move the WritePage method into guts_cli.

Reply:
Partially makes sense to me. 

Added `ClearPage` in surgeon package. But I still think it makes more sense to get both `ReadPage` and `WritePage` in package `guts_cli`.

Reply:
Both `ReadPage` and `WritePage` are more low-level, and it'd better to put them in `guts_cli`. While `CopyPage` and `ClearPage` is relatively high-level as compared to ReadPage & WritePage, so it's OK to put them in surgeon package. .

Reply:
I'm not proud of ReadPage & WritePage implementation to make them part of shared library. 

They are very inefficient in generic usage scenario - i.e. they shouldn't be used to read/write multiple pages. For this reason (and their simplicity) I though about them as private (yes: should be lower case) part of surgeon - not something for a wider consumption in other use-cases. 

Second idea is that some day guts_cli.go and page.go should get merged in a single file instead of 2 sibling implementations of the same data model. Thus I would prefer avoid further divergence.

Reply:
> Second idea is that some day guts_cli.go and page.go should get merged in a single file instead of 2 sibling implementations of the same data model. Thus I would prefer avoid further divergence.

Thanks @ptabor . This point totally makes sense to me. We need to refactor the design/implementation, of course in separate PRs.



> They are very inefficient in generic usage scenario - i.e. they shouldn't be used to read/write multiple pages. For this reason (and their simplicity) I though about them as private (yes: should be lower case) part of surgeon - not something for a wider consumption in other use-cases.

Yes, I am thinking the same thing, I mean the behavior of reading/writing single or multiple pages. It should be fine for now because the implementation of `ReadPage` and `WritePage` is Symmetrical. Both of them supporting operating multiple pages. 




> I'm not proud of ReadPage & WritePage implementation to make them part of shared library.

The implementation of `gut_cli` package has too much overlap with the existing `bbolt` package, we really need to merge them. It's the **next next** step we can do.

Reply:
If the `page.overflow > 0` , then it's dangerous to write only one of the pages. If we really need to do this, it definitely should be internal functions/methods. Let's investigate/do this separately.